test/test_stone.py

```diff
@@ -1315,6 +1445,340 @@ def test_forward_reference(self):
         self.assertEqual(
             api.namespaces['test'].data_types[0].fields[0].default.tag_name, 'a')
 
+    def test_struct_patch_semantics(self):
+        # Test patching normal struct
+        text = textwrap.dedent("""\
+            namespace files
+            struct QuotaInfo
+                "The space quota info for a user."
+                user_id String
+                    "The user associated with the quota."
+                example default
+                    user_id="1234"
+                example pro
+                    user_id="1234P"
+            patch struct QuotaInfo
+                quota UInt64
+                    "The user's total quota allocation (bytes)."
+                example default
+                    quota=2000000000
+                example pro
+                    quota=100000000000
+            """)
+        api = specs_to_ir([('test.stone', text)])
+
+        quota_info_dt = api.namespaces['files'].data_type_by_name['QuotaInfo']
+        self.assertEqual(quota_info_dt.fields[1].name, 'quota')
+        self.assertTrue(is_integer_type(quota_info_dt.fields[1].data_type))
+        self.assertEqual(quota_info_dt.fields[1].doc, "The user's total quota allocation (bytes).")
+        self.assertEqual(quota_info_dt.get_examples()['default'].value['quota'], 2000000000)
+        self.assertEqual(quota_info_dt.get_examples()['pro'].value['quota'], 100000000000)
+
+        # Test patching inherited struct
+        text = textwrap.dedent("""\
+            namespace files
+            struct QuotaInfo
+                "The space quota info for a user."
+                user_id String
+                    "The user associated with the quota."
+                example default
+                    user_id="1234"
+                example pro
+                    user_id="1234P"
+            struct QuotaInfoPersonal extends QuotaInfo
+                "The space quota info for a personal user."
+                personal_quota UInt64
+                    "The user's personal quota allocation (bytes)."
+            patch struct QuotaInfo
+                quota UInt64
+                    "The user's total quota allocation (bytes)."
+                example default
+                    quota=2000000000
+                example pro
+                    quota=100000000000
+            """)
+        api = specs_to_ir([('test.stone', text)])
+
+        quota_info_dt = api.namespaces['files'].data_type_by_name['QuotaInfoPersonal']
+        self.assertEqual(quota_info_dt.all_fields[1].name, 'quota')
+        self.assertTrue(is_integer_type(quota_info_dt.all_fields[1].data_type))
+        self.assertEqual(
+            quota_info_dt.all_fields[1].doc, "The user's total quota allocation (bytes).")
+
+        # Test patching enumerated supertype struct
```

Comment:
Testing patching the parent type of an enumerated subtype.

Reply:
Good idea. Done.

Reply:
Done.